pkg/services/star/starimpl/star_store.go

```diff
@@ -0,0 +1,80 @@
+package starimpl
+
+import (
+	"context"
+
+	"github.com/grafana/grafana/pkg/services/sqlstore"
+	"github.com/grafana/grafana/pkg/services/star"
+)
+
+type store interface {
+	isStarredByUserCtx(ctx context.Context, query *star.IsStarredByUserQuery) (bool, error)
+	insert(ctx context.Context, cmd *star.StarDashboardCommand) error
+	delete(ctx context.Context, cmd *star.UnstarDashboardCommand) error
+	getUserStars(ctx context.Context, query *star.GetUserStarsQuery) (map[int64]bool, error)
+}
+
+type storeImpl struct {
+	sqlStore sqlstore.Store
```

Comment:
We should have a narrower interface with `WithDbSession` and `WithTransactionalDbSession` only, no need to bring it complete store at this point.